src/openms/source/FORMAT/OMSFile.cpp

```diff
@@ -0,0 +1,2675 @@
+      Key value_id = storeDataValue_(info.getMetaValue(info_key));
+      query.bindValue(":data_value_id", value_id);
+      if (!query.exec())
+      {
+        raiseDBError_(query.lastError(), __LINE__, OPENMS_PRETTY_FUNCTION,
+                      "error inserting data");
+      }
+    }
+  }
+
+
+  void OMSFile::OMSFileStore::createTableAppliedProcessingStep_(
+    const String& parent_table)
+  {
+    createTable_(
+      parent_table + "_AppliedProcessingStep",
+      "parent_id INTEGER NOT NULL, "                                    \
+      "processing_step_id INTEGER, "                                    \
+      "processing_step_order INTEGER NOT NULL, "                        \
+      "score_type_id INTEGER, "                                         \
+      "score REAL, "                                                    \
+      "UNIQUE (parent_id, processing_step_id, score_type_id), "         \
+      "FOREIGN KEY (parent_id) REFERENCES " + parent_table + " (id), "  \
+      "FOREIGN KEY (score_type_id) REFERENCES ID_ScoreType (id), "      \
+      "FOREIGN KEY (processing_step_id) REFERENCES ID_ProcessingStep (id)");
+    // @TODO: add constraint that "processing_step_id" and "score_type_id"
+    // can't both be NULL
+    // @TODO: add constraint that "processing_step_order" must match "..._id"?
+    // @TODO: normalize table? (splitting into multiple tables is awkward here)
+  }
+
+
+  void OMSFile::OMSFileStore::storeAppliedProcessingStep_(
+    const ID::AppliedProcessingStep& step, Size step_order,
+    const String& parent_table, Key parent_id)
+  {
+    // this assumes the "..._AppliedProcessingStep" table exists already!
+    String table = parent_table + "_AppliedProcessingStep";
+
+    QSqlQuery query(QSqlDatabase::database(db_name_));
+    query.prepare("INSERT INTO " + table.toQString() + " VALUES ("  \
+                  ":parent_id, "                                    \
+                  ":processing_step_id, "                           \
+                  ":processing_step_order, "                        \
+                  ":score_type_id, "                                \
+                  ":score)");
+    query.bindValue(":parent_id", parent_id);
+    query.bindValue(":processing_step_order", int(step_order));
+    if (step.processing_step_opt)
+    {
+      query.bindValue(":processing_step_id",
+                      Key(&(**step.processing_step_opt)));
+      if (step.scores.empty()) // insert processing step information only
+      {
+        if (!query.exec())
+        {
+          raiseDBError_(query.lastError(), __LINE__, OPENMS_PRETTY_FUNCTION,
+                        "error inserting data");
+        }
+      }
+    } // else: use NULL for missing processing step reference
+    for (const auto& score_pair : step.scores)
+    {
+      query.bindValue(":score_type_id", Key(&(*score_pair.first)));
+      query.bindValue(":score", score_pair.second);
+      if (!query.exec())
```

Comment:
also here, do you really have to execute a query each time? I think it would be faster to use `execBatch` and insert many values at once